pkgs/development/ocaml-modules/rpclib/default.nix

```diff
@@ -0,0 +1,22 @@
+{ stdenv, fetchFromGitHub, buildDunePackage, alcotest, cmdliner, rresult, xmlm , yojson }:
+
+buildDunePackage rec {
+  pname = "rpclib";
+  version = "5.9.0";
+
+  src = fetchFromGitHub {
+    owner = "mirage";
+    repo = "ocaml-rpc";
+    rev = "v${version}";
+    sha256 = "108cw68kqzqbdbspdbdw8xrj186hrp0qz9kzslbsn0ls945641sy";
+  };
+
+  buildInputs = [ alcotest cmdliner rresult xmlm yojson ];
```

Comment:
Some of these libraries should be propagated. At least `rresult` and `result`; probably also `xmlm`.

Reply:
btw @vbgl, how do I know which deps should be propagated (in general)?

Reply:
Can you use package A without depending on package B? If the answer is “no”, then package A should propagate package B.

Regarding OCaml, the META files usually give precise information about link-time dependencies (look for “requires” lines in these files).

Reply:
thank you!